waspc/data/Generator/templates/server/src/jobs/pgBossJob.js

```diff
@@ -0,0 +1,70 @@
+import { boss } from './pgBoss.js'
+import { SubmittedJob } from './SubmittedJob.js'
+
+export const PG_BOSS_EXECUTOR_NAME = Symbol('PgBoss')
+
+class PgBossSubmittedJob extends SubmittedJob {
+  /**
+   * @param {string} jobName - The user-defined job name in their .wasp file.
+   * @param {string} jobId - A UUID for a given job in that executor's ecosystem.
+   * @param {string} executor - The name of the executor that is running the job.
+   * @param {object} pgBoss - PgBoss specific helpers, e.g., cancel()/resume()/details()
+   *                          NOTE: This param will see some flux, so not making a proper class right now.
+   */
+  constructor(jobName, jobId, executor, pgBoss) {
+    super(jobName, jobId, executor)
+    this.pgBoss = pgBoss
+  }
+}
+
+class PgBossJob {
+  constructor(values) {
+    this.jobFn = () => { }
+    this.defaultJobOptions = {}
+    this.startAfter = 0
+    this.jobName = 'unknown'
+    Object.assign(this, values)
+  }
+
+  delay(startAfter) {
+    return new PgBossJob({ ...this, startAfter })
+  }
+
+  /**
+   * Submits the job to PgBoss.
+   * @param {object} jobArgs - The job arguments supplied by the user for their callback.
+   * @param {string} jobOptions - PgBoss specific options for boss.send(), which can override their defaultJobOptions
+   *                              specified by their Wasp file.
+   */
+  async submit(jobArgs, jobOptions) {
+    const jobId = await boss.send(this.jobName, jobArgs, { ...this.defaultJobOptions, startAfter: this.startAfter, ...jobOptions })
+    return new PgBossSubmittedJob(this.jobName, jobId, PG_BOSS_EXECUTOR_NAME, {
+      async cancel() { return boss.cancel(jobId) },
+      async resume() { return boss.resume(jobId) },
+      async details() { return boss.getJobById(jobId) }
+    })
+  }
+}
+
+/**
+ * Initialized the PgBoss executor by registering this job function.
+ * @param {object} jobName - The user-defined job name in their .wasp file.
```

Comment:
We don't have to mention it explicitly, but it will end up being that. It is sort of important in the PgBoss world too, as that is the job name they will see if they go to the PgBoss schema.
<img width="1062" alt="Screen Shot 2022-04-27 at 8 32 33 AM" src="https://user-images.githubusercontent.com/523636/165518798-72a5e43e-d5fc-4428-9250-56a6d4f5bf01.png">
It will also have some future implications if they change the name of the job in their wasp file they may have stale scheduled jobs, so we will have to mention a recommended "cleaning process", etc.